addons/sourcemod/scripting/nd_bot_radio_helpers.sp

```diff
@@ -0,0 +1,466 @@
+    }
+
+    return Plugin_Handled;
+}
+
+public Action Timer_MedicChangeWeapon(Handle hTimer, any iUserId)
+{
+    int iClient = GetClientOfUserId(iUserId);
+    if (iClient && IsClientInGame(iClient))
+    {
+        FakeClientCommand(iClient, "use weapon_mp7");
+    }
+
+    return Plugin_Handled;
+}
+
+void ND_ClientLookAtTarget(int client, int target)
+{
+    float fGoalPosition[3];
+    GetClientEyePosition(target, fGoalPosition);
+
+    float fPosition[3];
+    GetClientEyePosition(client, fPosition);
+
+    // get normalised direction from target to client
+    float fVector[3];
+    MakeVectorFromPoints(fPosition, fGoalPosition, fVector);
+    float fDesiredAngles[3];
+    GetVectorAngles(fVector, fDesiredAngles);
+
+    // ease the current direction to the target direction
+    fDesiredAngles[0] = AngleNormalize(fDesiredAngles[0]);
+    fDesiredAngles[1] = AngleNormalize(fDesiredAngles[1]);
+    fDesiredAngles[2] = AngleNormalize(fDesiredAngles[2]);
+
+    TeleportEntity(client, NULL_VECTOR, fDesiredAngles, NULL_VECTOR);
+}
+
+bool CanClientSeeTarget(int Viewer, int Target, float fMaxDistance=0.0, float fThreshold=0.70)
+{
+    // Retrieve view and target eyes position
+    float fViewPos[3];
+    GetClientEyePosition(Viewer, fViewPos);
+    float fViewAng[3];
+    GetClientEyeAngles(Viewer, fViewAng);
+    float fViewDir[3];
+    float fTargetPos[3];
+    GetClientEyePosition(Target, fTargetPos);
+    float fTargetDir[3];
+    float fDistance[3];
+
+    // Calculate view direction
+    fViewAng[0] = fViewAng[2] = 0.0;
+    GetAngleVectors(fViewAng, fViewDir, NULL_VECTOR, NULL_VECTOR);
+
+    // Calculate distance to viewer to see if it can be seen.
+    fDistance[0] = fTargetPos[0] - fViewPos[0];
+    fDistance[1] = fTargetPos[1] - fViewPos[1];
+    fDistance[2] = 0.0;
+    if (fMaxDistance != 0.0)
+    {
+        if (((fDistance[0] * fDistance[0]) + (fDistance[1] * fDistance[1])) >= (fMaxDistance * fMaxDistance))
+        {
+            return false;
+        }
+    }
+
+    // Check dot product. If it's negative, that means the viewer is facing backwards to the target.
+    NormalizeVector(fDistance, fTargetDir);
+    if (GetVectorDotProduct(fViewDir, fTargetDir) < fThreshold)
+    {
+        return false;
+    }
+
+    return IsThereObstacleBetweenPoints(fViewPos, fTargetPos);
+}
+
+bool IsThereObstacleBetweenPoints(float fViewPos[3], float fTargetPos[3])
+{
+    // check if there are no obstacles in between through raycasting
+    Handle hTrace = TR_TraceRayFilterEx(fViewPos, fTargetPos, MASK_PLAYERSOLID_BRUSHONLY, RayType_EndPoint, ClientViewsFilter);
+    if (TR_DidHit(hTrace))
+    {
+        CloseHandle(hTrace);
+        return false;
+    }
+
+    // visible
+    CloseHandle(hTrace);
+    return true;
+}
+
+public bool ClientViewsFilter(int Entity, int Mask, any Junk)
+{
+    if (Entity >= 1 && Entity <= MaxClients)
+    {
+        return false;
+    }
+
+    return true;
+}
```

Comment:
This operation is more efficient when compiled to binary.
`return !(Entity >= 1 && Entity <= MaxClients);`

Reply:
The compiled binary is equivalent in both cases, so there are no efficiency gains by changing it. The readability is improved in this form.